src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1977,7 +1978,7 @@ protected void EnsureNullabilityAnalysisPerformedIfNecessary()
                 // Not all speculative models are created with existing snapshots. Attributes,
                 // TypeSyntaxes, and MethodBodies do not depend on existing state in a member,
                 // and so the SnapshotManager can be null in these cases.
-                if (_parentSnapshotManagerOpt is null)
+                if (_parentSnapshotManagerOpt is null || !isNullableAnalysisEnabled)
```

Comment:
>!isNullableAnalysisEnabled [](start = 57, length = 26)

It feels unintuitive to call ```rewriteAndCache``` when ```!isNullableAnalysisEnabled```. It looks like we are taking dependency on ```rewriteAndCache``` to not cache in this situation. Perhaps it is worth making that explicit, i.e. not relying on ```rewriteAndCache``` for that.